packages/@aws-cdk/aws-codepipeline-actions/test/integ.pipeline-ecs-create-cross-account.lit.ts

```diff
@@ -0,0 +1,221 @@
+     * const deployRoleName = `cdk-${cdkBootstrapQualifier}-deploy-role-${props.env!.account!}-${props.env!.region!}`;
+     *
+     * Ensure the role has permissions to deploy to ECS refer to.
+     * https://docs.aws.amazon.com/codepipeline/latest/userguide/how-to-custom-role.html#how-to-update-role-new-services
+     */
+    const deployArn = this.formatArn({
+      region: '',
+      service: 'iam',
+      resource: 'role',
+      resourceName: 'deployrole',
+    });
+    this.deployRole = iam.Role.fromRoleArn(
+      this,
+      'DeployRole',
+      deployArn,
+    );
+    // Adding ECS Deploy Permissions to deployRole
+    this.deployRole.addToPrincipalPolicy(
+      new iam.PolicyStatement({
+        actions: [
+          'ecs:DescribeServices',
+          'ecs:DescribeTaskDefinition',
+          'ecs:DescribeTasks',
+          'ecs:ListTasks',
+          'ecs:RegisterTaskDefinition',
+          'ecs:UpdateService',
+        ],
+        resources: ['*'],
+      }),
+    );
+
+    this.deployRole.addToPrincipalPolicy(
+      new iam.PolicyStatement({
+        actions: ['iam:PassRole'],
+        resources: ['*'],
+        conditions: {
+          StringEqualsIfExists: {
+            'iam:PassedToService': [
+              'ec2.amazonaws.com',
+              'ecs-tasks.amazonaws.com',
+            ],
+          },
+        },
+      }),
+    );
+
+  }
+}
+
+/**
+ * This is the Stage which does our create using {@link TestCreateEcsStack}.
+ */
+class TestCreateEcsStage extends cdk.Stage {
+  public readonly serviceArn: string;
+  public readonly clusterName: string;
+  public readonly deployRole: iam.IRole;
+
+  constructor(scope: Construct, id: string, props: cdk.StageProps) {
+    super(scope, id, props);
+    const testStack = new TestCreateEcsStack(this, 'ecsStack', {});
+    this.serviceArn = testStack.serviceArn;
+    this.clusterName = testStack.clusterName;
+    this.deployRole = testStack.deployRole;
+  }
+}
+
+/**
+ * This is our pipeline which will create an ECS Service and deploy
+ * to is using {@link EcsDeployAction} using our {@link TestCreateEcsStage}
+ */
+class PipelineEcsDeployCreateStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: cdk.StackProps) {
+    super(scope, id, props);
+    const artifact = new codepipeline.Artifact('Artifact');
+    const bucket = new s3.Bucket(this, 'PipelineBucket', {
+      versioned: true,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+    });
+    const source = new cpactions.S3SourceAction({
+      actionName: 'Source',
+      output: artifact,
+      bucket,
+      bucketKey: 'key',
+    });
+    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
+      stages: [
+        {
+          stageName: 'Source',
+          actions: [source],
+        },
+      ],
+    });
+    const testStage = new TestCreateEcsStage(
+      this,
+      'TestStage',
+      {
+        env: {
+          region: 'us-west-2',
+          account: '123456789012',
```

Comment:
Can we rename these to `'service-account`' / `'service-region'`, and the others used to `'pipeline-account'` / `'pipeline-region'`? Otherwise, reading the expected.json file is pretty damn difficult.

The same comment in the other example.